subsys/net/ip/tcp.c

```diff
@@ -1230,3 +1231,71 @@ struct net_buf *net_tcp_set_chksum(struct net_pkt *pkt, struct net_buf *frag)
 
 	return frag;
 }
+
+int tcp_parse_opts(struct net_pkt *pkt, int opt_totlen, u16_t *mss)
+{
+	struct net_buf *frag = pkt->frags;
+	u16_t pos = net_pkt_ip_hdr_len(pkt)
+		  + net_pkt_ipv6_ext_len(pkt)
+		  + sizeof(struct net_tcp_hdr);
+
+	u8_t opt, optlen;
+
+	while (opt_totlen) {
+		frag = net_frag_read(frag, pos, &pos, sizeof(opt), &opt);
```

Comment:
We need to check the return values here so that we now whether the packet is malformed or not.